src/chronify/time_series_mapper_base.py

```diff
@@ -142,4 +173,30 @@ def _apply_mapping(
 
     on_stmt = reduce(and_, (left_table.c[x] == right_table.c["from_" + x] for x in keys))
     query = select(*select_stmt).select_from(left_table).join(right_table, on_stmt)
-    create_table(to_schema.name, query, engine, metadata)
+
+    if output_file is not None:
+        write_query_to_parquet(engine, str(query), output_file, overwrite=True)
+        return
+
+    if engine.name == "hive":
+        create_materialized_view(
+            str(query), to_schema.name, engine, metadata, scratch_dir=scratch_dir
+        )
+    else:
+        create_table(to_schema.name, query, engine, metadata)
+
+
+def _create_view_if_necessary(
+    engine: Engine,
+    metadata: MetaData,
+    name: str,
+    check_mapped_timestamps: bool,
+    output_file: Optional[Path],
+) -> bool:
+    if check_mapped_timestamps:
+        if output_file is not None:
```

Comment:
So if output_file is None, no view is created? Wouldn't that fail check_timestamps() later?

Reply:
There is a bit of overlapping logic here. If output_file is None, we actually materialized the mapped result into a database table. If output is not None, the mapped result is in that file and nothing is in the database. We have to create a view to check the timestamps.